src/qibolab/transpilers/general_connectivity.py

```diff
@@ -0,0 +1,411 @@
+        """Initialize the transpiled circuit
+
+        Args:
+            Args: qibo_circuit (qibo.Circuit): circuit to be transpiled.
+        """
+        nodes = self._connectivity.number_of_nodes()
+        qubits = qibo_circuit.nqubits
+        if qubits > nodes:
+            raise_error(ValueError, "There are not enough physical qubits in the hardware to map the circuit")
+        elif qubits == nodes:
+            new_circuit = Circuit(nodes)
+        else:
+            log.warning(
+                "You are using more physical qubits than required by the circuit, some qubits will be added to the circuit"
+            )
+            new_circuit = Circuit(nodes)
+        self.transpiled_circuit = new_circuit
+
+    def init_mapping_circuit(self, circuit, qubit_map):
+        """Initial qubit mapping of the transpiled qibo circuit
+
+        Args:
+            circuit (qibo.Circuit): transpiled qibo circuit.
+            qubit_map (np.array): initial qubit mapping.
+
+        Returns:
+            new_circuit (qibo.Circuit): transpiled circuit mapped with initial qubit mapping.
+        """
+        new_circuit = Circuit(self._connectivity.number_of_nodes())
+        for gate in circuit.queue:
+            new_circuit.add(gate.on_qubits({q: qubit_map[q] for q in gate.qubits}))
+        return new_circuit
+
+    def add_gates(self, qibo_circuit, matched_gates):
+        """Add one and two qubit gates to transpiled circuit until connectivity is matched
+
+        Args:
+            qibo_circuit (qibo.Circuit): circuit to be transpiled.
+            matched_gates (int): number of two qubit gates that can be applied with the current qubit mapping
+        """
+        index = 0
+        while self._circuit_position < len(qibo_circuit.queue):
+            gate = qibo_circuit.queue[self._circuit_position]
+            if len(gate.qubits) == 1:
+                self.transpiled_circuit.add(gate.on_qubits({gate.qubits[0]: self._qubit_map[gate.qubits[0]]}))
+                self._circuit_position += 1
+            else:
+                index += 1
+                if index == matched_gates + 1:
+                    break
+                else:
+                    self.transpiled_circuit.add(
+                        gate.on_qubits(
+                            {
+                                gate.qubits[0]: self._qubit_map[gate.qubits[0]],
+                                gate.qubits[1]: self._qubit_map[gate.qubits[1]],
+                            }
+                        )
+                    )
+                    self._circuit_position += 1
+
+    def add_swaps(self, path, meeting_point):
+        """Add swaps to the transpiled circuit to move qubits
+
+        Args:
+            path (list): path to move qubits.
+            meeting_point (int): qubit meeting point in the path.
+        """
+        forward = path[0 : meeting_point + 1]
+        backward = path[meeting_point + 1 :]
+        backward.reverse()
```

Comment:
```suggestion
        backward = path[meeting_point + 1 ::-1]
```
Maybe this is even too much, in the end it is more or less the same (but, personally, I don't like too much the calls that do not return their result...).